### PR TITLE
feat: add system metrics enabled config

### PR DESCRIPTION
**Pull request by thugrock7**
## Description
Config to enable system metrics. These system metrics are setup in goagent
Linked Issue: https://github.com/hypertrace/goagent/pull/235

### CODE REVIEW

File: proto/hypertrace/agent/config/v1/config.proto
Comment (varkey98) [inline proto/hypertrace/agent/config/v1/config.proto:199]: lets move this and the above enabled flag to a common metrics message and deprecate the metrics_enabled one

**Comment by varkey98** [comment]
Static check's been failing here too @tim-mwangi  :|

**Comment by tim-mwangi** [comment]
> Static check's been failing here too @tim-mwangi :|

I wonder if it's a permissions thing. I have added @thugrock7 to the collaborators with Write access. I will do the same for the goagent repo. It's better if @thugrock7 asked to be added to the Hypertrace organization.

File: proto/hypertrace/agent/config/v1/config.proto
Comment (varkey98) [inline proto/hypertrace/agent/config/v1/config.proto:198]: need to add the Metrics message object here as well.

File: proto/hypertrace/agent/config/v1/config.proto
Comment (varkey98) [inline proto/hypertrace/agent/config/v1/config.proto:205]: just `enabled` would be fine

Comment (thugrock7) [reply proto/hypertrace/agent/config/v1/config.proto:205]: keeping just `enabled`, is creating env var as `HT_ENABLED`, making it like this creates `HT_METRICS_ENABLED`, so named it like this. what do you suggest @varkey98 ?

Comment (varkey98) [reply proto/hypertrace/agent/config/v1/config.proto:205]: @thugrock7 resolve [this](https://github.com/hypertrace/agent-config/pull/163#discussion_r1677519678) comment and it'll get fixed. It's happening because this message is a dangling one and not referred by anything else

File: proto/hypertrace/agent/config/v1/config.proto
Comment (varkey98) [inline proto/hypertrace/agent/config/v1/config.proto:216]: nit: update the comment to `The system metrics is only supported in goagent as of now`

**Review by varkey98 — APPROVED** [review]
LGTM, pls wait for @tim-mwangi 's  or @ryanericson 's approval

File: proto/hypertrace/agent/config/v1/config.proto
Comment (shashank11p) [reply proto/hypertrace/agent/config/v1/config.proto:216]: `Whether to capture system metrics or not. The system metrics are setup in each agent`
We do not need to add that it's only supported in goagent, because we can update any agent at any time. With this config, any agent can use it.
We can keep this as part of this PR that during the time of this PR, we were only adding support for goagent.

Comment (varkey98) [reply proto/hypertrace/agent/config/v1/config.proto:216]: You're right, just wanted to propagate the idea that support needs to be added explicitly

**Review by tim-mwangi — APPROVED** [review]

File: proto/hypertrace/agent/config/v1/config.proto
Comment (varkey98) [reply proto/hypertrace/agent/config/v1/config.proto:208]: Yes, for ebpf we have it on by default. Was wondering from a generic metrics config point of view, if we need to make it modular (coming from our discussion [here](https://github.com/Traceableai/traceable-agent/pull/2032#discussion_r1548657796)).

Comment (varkey98) [reply proto/hypertrace/agent/config/v1/config.proto:208]: But that ship has already sailed as the levels of this config is different in both the places. Maybe we can merge both of these protos into one and make a single config, but ig that'd require the hypertrace agents to be deprecated :|

Comment (ryanericson) [reply proto/hypertrace/agent/config/v1/config.proto:208]: In that case I think it can all just be controlled by this single flag then. So no need for this change.

**Comment by varkey98** [comment]
closing this pr as mentioned here, https://github.com/hypertrace/agent-config/pull/163#discussion_r1679999348

**Pull request closed without merging**
